src/include/heating_spot_foil.hxx

```diff
@@ -11,8 +11,9 @@ struct HeatingSpotFoilParams
   double xc;
   double yc;
   double rH;
-  double T;
+  double T[10];
   double Mi;
+  int n_kinds;
```

Comment:
Looks good, except that since the number 10 shows up in a bunch of places later, it'd be good to use a named constant for it (e.g., `#define HEATING_MAX_N_KINDS (10)`, or something fancier C++-y if you prefer.)